server/src/com/cloud/resourcelimit/ResourceLimitManagerImpl.java

```diff
@@ -938,11 +942,28 @@ public long countCpusForAccount(long accountId) {
         sc.setJoinParameters("offerings", "state", new Object[] {State.Destroyed, State.Error, State.Expunging});
         sc.setJoinParameters("offerings", "displayVm", 1);
         List<SumCount> cpus = _serviceOfferingDao.customSearch(sc, null);
+        long total = 0;
         if (cpus != null) {
-            return cpus.get(0).sum;
-        } else {
-            return 0;
+            total = cpus.get(0).sum;
+        }
+
+        GenericSearchBuilder<UserVmDetailVO, SumCount> detailCpuSearch = _userVmDetailsDao.createSearchBuilder(SumCount.class);
+        detailCpuSearch.select("sum", Func.SUM, detailCpuSearch.entity().getValue());
+        detailCpuSearch.and("name", detailCpuSearch.entity().getName(), Op.EQ);
+        detailCpuSearch.join("vms", join1, detailCpuSearch.entity().getResourceId(), join1.entity().getId(),  JoinBuilder.JoinType.INNER);
+
+        sc = detailCpuSearch.create();
+        sc.setParameters("name", "cpu");
+        sc.setJoinParameters("offerings", "accountId", accountId);
+        sc.setJoinParameters("offerings", "type", VirtualMachine.Type.User);
+        sc.setJoinParameters("offerings", "state", new Object[] {State.Destroyed, State.Error, State.Expunging});
+        sc.setJoinParameters("offerings", "displayVm", 1);
```

Comment:
It's an arbitrary name to reference when assigning where clause parameters below that. The join is on vm_instance treacle ultimately, so that's suitable shorthand.

Reply:
Crap, I didn't change it below to vms